src/pymoca/backends/casadi/generator.py

```diff
@@ -596,43 +596,51 @@ def get_python_type(tree):
             return float
 
     def get_shape(self, tree):
-        return [self.get_integer(d) for d in tree.dimensions]
+        return [[self.get_integer(d) for d in d_list] for d_list in tree.dimensions]
 
     def get_symbol(self, tree):
         # Create symbol
         shape = self.get_shape(tree)
 
-        if any(isinstance(x, slice) for x in shape):
+        if any(any(isinstance(x, slice) for x in var_shape) for var_shape in shape):
             # Symbol has unspecified dimensions. Value is specified, and
             # carries the correct dimensions.
 
             # We should only get slices as dimensions for a symbol if one of
             # the dimensions is unspecified, i.e. None.
-            assert None in itertools.chain.from_iterable((x.start, x.stop) for x in shape if isinstance(x, slice))
+            assert None in (itertools.chain.from_iterable((x.start, x.stop)
+                            for var_shape in shape for x in var_shape if isinstance(x, slice)))
 
             val_shape = np.array(self.src[tree.value]).shape
 
             # Check if specified dimensions agree between definition and value
-            for dim_i, dim_size in enumerate(shape):
-                if isinstance(dim_size, slice):
-                    continue
-
-                if val_shape[dim_i] != dim_size:
-                    raise Exception("Dimension {} of definition and value for symbol {} differs: {} != {}".format(
+            val_dim_i = -1
+            for var_i, var_shape in enumerate(shape):
+                for dim_i, dim_size in enumerate(var_shape):
+                    if dim_size is None:
+                        continue
+
+                    val_dim_i += 1
+                    if isinstance(dim_size, slice):
+                        shape[var_i][dim_i] = val_shape[val_dim_i]
+                        continue
+
+                    if val_shape[val_dim_i] != dim_size:
+                        raise Exception("Dimension {} of definition and value for symbol {} differs: {} != {}".format(
                         dim_i + 1, tree.name, dim_size, val_shape[dim_i]))
```

Comment:
The error message should also print `val_shape[val_dim_i]`, not `val_shape[dim_i]`. Also, `dim_i + 1` might not really make sense anymore either. Would be nice to rethink this  error message such that it provides useful information in the new way of handling things.